game-core/src/main/java/games/strategy/triplea/delegate/AbstractEndTurnDelegate.java

```diff
@@ -516,7 +517,7 @@ public String getDisplayName() {
   private static Comparator<Territory> getSingleNeighborBlockadesThenHighestToLowestProduction(
       final Collection<Territory> blockadeZones, final GameMap map) {
     return (t1, t2) -> {
-      if (t1 == t2 || (t1 == null && t2 == null)) {
+      if (Objects.equals(t1, t2)) {
```

Comment:
Now that a normal equals is being used, a simple Comparator.comparing lambda can be used, there are 2 static methods that provide null compatible comparators.
Sorry for not realising that earlier.